src/main/java/com/growup/pms/status/controller/StatusControllerV1.java

```diff
@@ -24,13 +27,14 @@
 @Slf4j
 @RestController
 @RequiredArgsConstructor
-@RequestMapping("/api/v1/status")
+@RequestMapping("/api/v1/project")
```

Comment:
`projectId`가 누락된 것 같아요! 추후 구현에서 반영해주시면 감사하겠습니다 :smile: 

```suggestion
@RequestMapping("/api/v1/project/{projectId}/status")
```

Reply:
아 모든 매핑에서 받으라는 말씀이셨군요ㅎ 사용하지 않는 데이터라서 사용하는 API 에서만 `@PathVariable` 로 받으려고 했는데, 그래도 위와 같이 변경하는게 더 나으려나요?

Reply:
프로젝트마다 상태를 별도로 관리할 수 있는 것으로 알고 있었습니다! 그래서 프로젝트라는 컬렉션의 하위 컬렉션으로 상태가 들어가고 있어서 API 명세서에 `/api/v1/project/{projectId}/status`를 공통 경로로 가져가는 것으로 알고 있는데, 여기서 depth가 증가하면 영환님이 저번에 쓰신 대로 `/api/v1/status/{statusId}/...`로 변경해도 괜찮을 것 같아요!

추가로 [웹 API 디자인 모범 사례](https://learn.microsoft.com/ko-kr/azure/architecture/best-practices/api-design)를 아래에다 덧붙여봅니다! `/customers/5/orders/3/items`와 같이 `컬렉션/항목/컬렉션/항목/컬렉션`처럼 과도하게 확장되는 경우에는 `/status/{statusId}/...`를 고민해보는 것도 좋은 것 같습니다! :smile: 

> [!NOTE]
> 서로 다른 리소스 형식과 이러한 연결을 표시하는 방법 사이의 관계도 고려해야 합니다. 예를 들어 /customers/5/orders는 고객 5에 대한 모든 주문을 나타낼 수 있습니다. 반대 방향으로 이동하여 /orders/99/customer 같은 URI를 사용하여 주문에서 고객으로의 연결을 표시할 수도 있습니다. 그러나 이 모델을 너무 많이 확장하면 구현이 어려울 수 있습니다. HTTP 응답 메시지의 본문에 연결된 리소스에 대한 탐색 가능한 링크를 제공하는 방법이 더 좋습니다. ... 리소스 URI를 컬렉션/항목/컬렉션보다 더 복잡하게 요구하지 않는 것이 좋습니다.

순서 변경이 결국 상태 리소스의 속성을 수정하는 것이다 보니 `editStatus(...)`에서 처리하는 것도 괜찮겠다 싶더라구요! 

![image](https://github.com/user-attachments/assets/f2ba6dbc-0be4-46ab-859c-698f0a560dfb)

```java
@RequestMapping("/api/v1/project/{projectId}/status")
public class StatusControllerV1 {
    @PostMapping
    public ResponseEntity<StatusResponse> createStatus(@PathVariable Long projectId, @Valid @RequestBody StatusCreateRequest request) {
        // ...
    }

    @GetMapping
    public ResponseEntity<PageResponse<List<StatusResponse>>> getStatuses(@PathVariable Long projectId) {
        // ...
    }

    @PatchMapping("/{statusId}")
    public ResponseEntity<Void> editStatus(@PathVariable Long projectId, @PathVariable Long statusId, @Valid @RequestBody StatusEditRequest request) {
        // ...
    }

    @DeleteMapping("/{statusId}")
    public ResponseEntity<Void> deleteStatus(@PathVariable Long projectId, @PathVariable Long statusId) {
        // ...
    }
}
```

Reply:
아... AOP를 통해서 프로젝트 권한 검사 시에 메서드 파라미터에 `projectId`가 있어야 해서 추가되어야 하는 것 같습니다... :disappointed:

```java
@Before("@annotation(requirePermission) && args(projectId, ..)")
public void checkProjectPermission(RequirePermission requirePermission, Long projectId) {
    checkPermission(requirePermission, projectUserRepository.getPermissionsForProjectUser(projectId, getCurrentUser().getId()));
}
```

Reply:
네 경로 부분은 좀 더 고민을 해봐야겠네요 😥
말씀하신 변경과 순서 변경으로 나눈 이유는 고민을 해보니, 정렬 순서만 변경하는 API 의 호출이 더 잦을 것 같은데, 이때마다 모든 데이터를 들고 이동하는게 불필요하지 않을까 싶어서 이렇게 분리를 해봤습니다!

Reply:
아 이 부분을 놓쳤네요.. 다음 기능 개발 시작 전에 반영해두도록 하겠습니다!

Reply:
순서 변경 시에 아래와 같이 요청이 날아가는 것으로 알고 있었습니다! 부분 변경이라 `sortOrder`만 날아가는 것으로 알고 있었는데, 여기서 혹시 제가 놓친 부분이 있을까요?

```http
PATCH /api/v1/status/{statusId}
Content-Type: application/json
...

{
    "sortOrder": 3 // sortOrder를 3으로 부분 변경
}
```

Reply:
네 제대로 이해하셨습니다. 순서 변경이 아니라 그냥 상태에 대한 변경에 포함시키는 경우 `name, colorCode` 이렇게 두 개의 값도 같이 요청에 담아서 보내야하고, 상태의 `name, colorCode` 부분만 변경하는 경우에도 `sortOrder` 를 같이 담아서 보내야해서 분리를 해봤습니다. 두 변경이 함께 이루어지는 경우가 별로 없을 것 같아서 분리하는게 낫지 않을까 하고 생각해봤어요 🤔

Reply:
만약에 `PATCH /api/v1/status/status{id}`라면, 아래와 같이 전송되는 것이 아닌

```json
{
    "name": "To Do",
    "colorCode": "334455", 
    "sortOrder": 4
}
```

아래와 같이 날아가는 것으로 알고 있었는데 혹시 맞을까요?

```json
{
    "sortOrder": 4
}
```

![image](https://github.com/user-attachments/assets/91c9a59b-8cf5-4abd-985c-8c8ae3bc0a43)

Reply:
`@PatchMapping` 구현 시에 이렇게 구현했었어서 혹시 참고삼아 PR을 링크해봅니다 :cat: 

- #63

Reply:
`@PatchMapping` 에 `sortOrder`만 요청에 담아서 보내면
``` json
{
   "name": null,
   "colorCode": null,
   "sortOrder": 3
}
```
이렇게 날라가지 않나요? 제가 해당 부분에 대해서는 자세히 아는게 없네요 😥
근데 다시 보니까 `sortOrder` 를 `@RequestBody`를 사용해서 받아야 하는데 `@RequestParam` 으로 받고 있어서 수정을 해야하긴 하네요..

Reply:
PATCH에서 null을 명시적으로 보내면 그 값을 제거한다는 의미로 알고 있었습니다! 저도 PATCH에 대해서 자세히 알지는 못해서 우선 찾은 정보를 아래에다 추가적으로 공유하도록 하겠습니다 :cat: 

> 그렇다면 이번에는 PATCH의 경우를 살펴보자.
> 처음 들었던 예시로 다시 돌아가서 지하철노선(Line)이 가지고 있는 네 가지 필드 중 이번에는 이름(name)과 배차간격(intervalTime) 필드만 변경이 가능하다고 가정해보겠다. 클라이언트에서는 기존처럼 모든 필드를 요청에 실을 필요없이 아래처럼 name과 intervalTime에 대한 정보만 body에 담아서 보내면 된다.
>
> ...
> body: JSON.stringify({
>  name: "변경된 노선 이름",
>  intervalTime: 100
>})
>...
>PUT의 경우, URL에 담긴 정보로 엔티티를 식별할 수 없다면 생성까지 해야하기 때문에 엔티티에 필요한 모든 정보를 payload에 실어 보내야 한다.

이 내용은 Tecoble의 [자원을 수정하는 HTTP 메서드 - PUT vs PATCH](https://tecoble.techcourse.co.kr/post/2020-08-17-put-vs-patch/)에서 확인할 수 있었습니다! 전체 필드를 보내야 하는 경우에는 `PUT`을 고려해볼 수 있고, 부분 변경만 필요한 경우에는 `PATCH`인 것 같기도 합니다 :thinking:

Reply:
`PUT` 요청을 보냈을 때 누락된 필드가 있으면 해당 필드는 `null` 로 변경되어 저장이 되는 것으로 알고 있어요! 부분 변경을 하는 경우에 `PATCH` 를 사용하는 것으로 저도 알고 있습니다 😀
이 부분은 우선 두 AP를 통합해서 작성을 해두고, 차후에 기능을 구현 하면서 제가 테스트를 한번 해보겠습니다! 🥸

Reply:
항상 수고가 많으십니다! 저도 내일까지 가능하면 사용자 검색까지 마치도록 하겠습니다 :+1:

Reply:
넵 항상 꼼꼼하게 코드리뷰 해주셔서 감사합니다! ☺️